src/main/java/games/strategy/util/memento/PropertyBagMemento.java

```diff
@@ -0,0 +1,84 @@
+package games.strategy.util.memento;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Implementation of {@link Memento} that represents the originator state as a collection of name-value pairs.
+ *
+ * <p>
+ * Instances of this class are immutable.
+ * </p>
+ */
+public final class PropertyBagMemento implements Memento {
```

Comment:
I believe you realized this already, but in this case, `PropertyBag` refers to the memento's implementation rather than what is being "mementoed" (sorry for the made-up word :smile:).

In a previous comment, you had suggested avoiding the generic mapping of class fields to name-value pairs, so I have a note to myself to potentially replace `PropertyBagMemento` with an internally strongly-typed `GameDataMemento`.  I was still planning on pursuing that as the framework is able to persist more and more `GameData` fields.  So, I'm going to refrain from focusing on `PropertyBagMemento` too much at this time until the decision to abandon or not abandon it becomes clearer.

I agree with your comments about only having a single implementation of `Memento`.  And this may go back to instead using `Object` to type a memento instead of a marker interface.  Regardless, I think it's pretty clear that by the end of this feature, there will only be a single type of memento (for `GameData`), and the general abstraction represented by the interface will not be necessary.